modules/runner/src/main/java/org/apache/ignite/internal/configuration/storage/DistributedConfigurationStorage.java

```diff
@@ -111,88 +120,95 @@ public DistributedConfigurationStorage(MetaStorageManager metaStorageMgr, VaultM
+        return supplyAsync(() -> {
+            var data = new HashMap<String, Serializable>();
 
-        var rangeStart = new ByteArray(DISTRIBUTED_PREFIX + prefix);
+            var rangeStart = new ByteArray(DISTRIBUTED_PREFIX + prefix);
 
-        var rangeEnd = new ByteArray(incrementLastChar(DISTRIBUTED_PREFIX + prefix));
+            var rangeEnd = new ByteArray(incrementLastChar(DISTRIBUTED_PREFIX + prefix));
 
-        try (Cursor<Entry> entries = metaStorageMgr.range(rangeStart, rangeEnd)) {
-            for (Entry entry : entries) {
-                ByteArray key = entry.key();
-                byte[] value = entry.value();
+            try (Cursor<Entry> entries = metaStorageMgr.range(rangeStart, rangeEnd)) {
+                for (Entry entry : entries) {
+                    ByteArray key = entry.key();
+                    byte[] value = entry.value();
 
-                if (entry.tombstone()) {
-                    continue;
-                }
+                    if (entry.tombstone()) {
+                        continue;
+                    }
 
-                // Meta Storage should not return nulls as values
-                assert value != null;
+                    // Meta Storage should not return nulls as values
+                    assert value != null;
 
-                if (key.equals(MASTER_KEY)) {
-                    continue;
-                }
+                    if (key.equals(MASTER_KEY)) {
+                        continue;
+                    }
 
-                String dataKey = key.toString().substring(DISTRIBUTED_PREFIX.length());
+                    String dataKey = key.toString().substring(DISTRIBUTED_PREFIX.length());
 
-                data.put(dataKey, ConfigurationSerializationUtil.fromBytes(value));
+                    data.put(dataKey, ConfigurationSerializationUtil.fromBytes(value));
+                }
+            } catch (Exception e) {
+                throw new StorageException("Exception when closing a Meta Storage cursor", e);
             }
-        } catch (Exception e) {
-            throw new StorageException("Exception when closing a Meta Storage cursor", e);
-        }
 
-        return data;
+            return data;
+        }, threadPool);
     }
 
     /** {@inheritDoc} */
     @Override
-    public Serializable readLatest(String key) throws StorageException {
-        try {
-            Entry entry = metaStorageMgr.get(new ByteArray(DISTRIBUTED_PREFIX + key)).join();
+    public CompletableFuture<Serializable> readLatest(String key) {
+        return metaStorageMgr.get(new ByteArray(DISTRIBUTED_PREFIX + key))
+                .thenApply(entry -> {
+                    byte[] value = entry.value();
 
-            return entry.value() == null ? null : ConfigurationSerializationUtil.fromBytes(entry.value());
-        } catch (Exception e) {
-            throw new StorageException("Exception while reading data from Meta Storage", e);
```

Comment:
Should exception (if occurs) be wrapped in `StorageException` in the new code as well?